src/components/CustomInput.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <div class="custom-input">
+    <input 
+      id="answer" 
+      class="custom-input__button" 
+      :type="type" 
+      @model="selectedOption" 
+      @change="selectedAnswer"
+    > 
+    <label for="answer" :class="['custom-input__label', {'custom-input__label--checkbox' :type === 'checkbox','custom-input__label--radio' :type === 'radio'}]">
```

Comment:
and you can change `for="answer"` to `:for="name"` - which will give it the same dynamic ID as the one you set above.